android/src/toga_android/dialogs.py

```diff
@@ -130,25 +132,39 @@ def __init__(
 
 
 class OpenFileDialog(BaseDialog):
+    class HandlerOpenDialog(uf.HandlerFileDialog):
```

Comment:
Unless there's a reason for nesting classes like this, we generally keep classes at the module level.

Reply:
Here is the reason for the class to be used only with OpenFileDialog

Reply:
That's not a reason to nest classes. It complicates the class definition, and makes debugging harder because there's implicit context (and potentially closures) that can contribute to any bug being observed.